subsys/bluetooth/host/conn.c

```diff
@@ -1887,14 +1888,21 @@ struct bt_conn *bt_conn_ref(struct bt_conn *conn)
 
 void bt_conn_unref(struct bt_conn *conn)
 {
-	atomic_val_t old = atomic_dec(&conn->ref);
+	atomic_val_t old;
+
+	if (IS_ENABLED(CONFIG_BT_ISO) && conn->type == BT_CONN_TYPE_ISO &&
+	    atomic_get(&conn->ref) == 1) {
+		bt_iso_cleanup(conn);
+	}
+
+	old = atomic_dec(&conn->ref);
```

Comment:
I assume the reason for this change, is that we need to call `bt_iso_cleanup` with the ref "intact" for some reason. Could you possibly add a comment stating why we need this special handling for ISO channels?

Reply:
Yep, @joerchan pointed out that if we release the reference another thread may, in the meantime, allocated while bt_iso_cleanup has not completed it.

Reply:
Done.